### PR TITLE
Add steps for Frida snippet generation from JADX

### LINKED ISSUE #2070: [Tool] Add jadx to Frida Integration
https://twitter.com/leonjza/status/1495445380292558857?t=o7MgyV0wjMTQe2CLPrie4w&s=19

### PR DESCRIPTION
**Pull request by cgarst**
Detailing a feature added in JADX 1.3.3 where basic Frida scripts to capture method entry/exit parameters can be dynamically generated through the decompiler's graphical code browser.

--------------

Thank you for submitting a Pull Request to the OWASP MASTG. Please make sure that:

- [✓] Your contribution is written in the 2nd person (e.g. you)
- [✓] Your contribution is written in an active present form for as much as possible.
- [✓] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to MITRE/MASVS/etc. are up to date)
- [✓] Your contribution has proper formatted markdown and/or code
- [✓] Any references to website have been formatted as [TEXT](URL “NAME”)
- [✓] You verified/tested the effectiveness of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

This PR closes #2070

### CODE REVIEW

File: Document/0x08a-Testing-Tools.md [hunk 1]
```diff
@@ -211,6 +211,24 @@ Java.perform(function () {
 
 The above script calls `Java.perform` to make sure that your code gets executed in the context of the Java VM. It instantiates a wrapper for the `android.app.Activity` class via `Java.use` and overwrites the `onResume` function. The new `onResume` function implementation prints a notice to the console and calls the original `onResume` method by invoking `this.onResume` every time an activity is resumed in the app.
 
+The [JADX decompiler](https://github.com/skylot/jadx) can generate Frida snippets through it's graphical code browser which print method entry and exit parameters via a `Java.use` call. To use this feature, open the APK or DEX with `jadx-gui`, browse to the target method, right click the method name, and select "Copy as frida snippet (f)". For example using the MASTG `UnCrackable-Level1.apk` app:
```
Comment (cpholguera) [inline Document/0x08a-Testing-Tools.md:214]:
```suggestion
The [JADX decompiler](#jadx) (v1.3.3 and above) can generate Frida snippets through its graphical code browser. To use this feature, open the APK or DEX with `jadx-gui`, browse to the target method, right click the method name, and select "Copy as frida snippet (f)". For example using the MASTG [UnCrackable App for Android Level 1](0x08b-Reference-Apps.md#android-uncrackable-l1):
```

File: Document/0x08a-Testing-Tools.md [hunk 2]
```diff
@@ -211,6 +211,24 @@ Java.perform(function () {
...
+
+<img src="Images/Chapters/0x08a/jadx_copy_frida_snippet.png" width="100%" />
+
+The above steps place the following output in pasteboard to be pasted in a `.js` file and fed into `frida -U -l`.
```
Comment (cpholguera) [inline Document/0x08a-Testing-Tools.md:218]:
```suggestion
The above steps place the following output in the pasteboard, which you can then paste in a JavaScript file and feed into `frida -U -l`.
```

File: Document/0x08a-Testing-Tools.md [hunk 3]
```diff
@@ -211,6 +211,24 @@ Java.perform(function () {
...
+
+```java
```
Comment (cpholguera) [inline Document/0x08a-Testing-Tools.md:220]:
```suggestion
```javascript
```

File: Document/0x08a-Testing-Tools.md [hunk 4]
```diff
@@ -211,6 +211,24 @@ Java.perform(function () {
...
+let a = Java.use("sg.vantagepoint.a.a");
+a["a"].implementation = function (bArr, bArr2) {
+    console.log('a is called' + ', ' + 'bArr: ' + bArr + ', ' + 'bArr2: ' + bArr2);
+    let ret = this.a(bArr, bArr2);
+    console.log('a ret value is ' + ret);
+    return ret;
+};
+```
+
+The above code generated by `jadx-gui` hooks the `a` method within the `sg.vantagepoint.a.a` class and logs it's entry parameters and return values.
```
Comment (cpholguera) [inline Document/0x08a-Testing-Tools.md:230]:
```suggestion
The above code hooks the `a` method within the `sg.vantagepoint.a.a` class and logs its input parameters and return values.
```

**Review by cpholguera — CHANGES_REQUESTED** [review]
Just a couple of corrections and we're ready to go. Thank you @cgarst!

**Comment by cgarst** [comment]
Fixed, thanks!

**Review by cpholguera — APPROVED** [review]
Thanks a lot @cgarst!

**Pull request merged**